src/questFolder/gatherScrap.py

```diff
@@ -191,21 +191,25 @@ def generateDutyQuest(beUsefull,character,currentRoom):
 
                     if source is None and not character.getTerrain().scrapFields:
                         continue
-                    if src.quests.questMap["ClearInventory"].generateDutyQuest(beUsefull,character,room):
-                        beUsefull.idleCounter = 0
-                        return True
+
+                    step =  src.quests.questMap["ClearInventory"].generateDutyQuest(beUsefull,character,room)
+                    if step != (None,None):    
+                        if not dryRun:
+                            beUsefull.idleCounter = 0
+                        return step
 
                     if source:
                         pos = source[0]
                     else:
                         pos = random.choice(character.getTerrain().scrapFields)
 
-                    beUsefull.addQuest(src.quests.questMap["RestockRoom"](toRestock="Scrap",reason="fill scrap inputs"))
-                    beUsefull.addQuest(src.quests.questMap["GoToTile"](targetPosition=(room.xPosition,room.yPosition)))
-                    beUsefull.addQuest(src.quests.questMap["GatherScrap"](targetPosition=pos))
-                    beUsefull.addQuest(src.quests.questMap["GoToTile"](targetPosition=pos))
-                    beUsefull.idleCounter = 0
-                    return True
+                    quests = [src.quests.questMap["RestockRoom"](toRestock="Scrap",reason="fill scrap inputs"),
```

Comment:
i feel formatting like this would be easier

quests = [
                    src.quests.questMap["RestockRoom"](toRestock="Scrap",reason="fill scrap inputs"),
                    src.quests.questMap["GoToTile"](targetPosition=(room.xPosition,room.yPosition)),
                    src.quests.questMap["GatherScrap"](targetPosition=pos),
                    src.quests.questMap["GoToTile"](targetPosition=pos)
             ]